game-core/src/main/java/games/strategy/triplea/ui/MovePanel.java

```diff
@@ -1220,6 +1196,23 @@ private void updateUnitsThatCanMoveOnRoute(final Collection<Unit> units, final R
     unitsThatCanMoveOnRoute = new ArrayList<>(bestWithDependents);
   }
 
+  private Tuple<MoveValidationResult, List<Unit>> validateMoveWithDependents(
+      final List<Unit> units, final Map<Unit, Unit> unitsToTransports, final Route route) {
+    final List<Unit> bestWithDependents = addMustMoveWith(units);
+    final MoveValidationResult result;
+    getData().acquireReadLock();
```

Comment:
It was under a read lock in the first case, but not in the other cases.

Reply:
I'm skeptical of the read lock usages. If we go to `LockUtil`, it basically says the whole thing is a hack: https://github.com/triplea-game/triplea/blob/master/game-core/src/main/java/org/triplea/thread/LockUtil.java#L22

In the case above, if the data changes as soon as we release the lock, will the result be incorrect? In that case, will the lock have actually helped anything?

IMO this is a problem with most locks in TripleA, they are not actually held long enough to be effective. On the other hand, some of the locks in TripleA are likely to be necessary, and we do not know which ones are, so we can't just remove them.

@asvitkine what are your thoughts, better to have unified (but unnecessary locking perhaps), or is this locking probably more clearly not needed?

Reply:
In this case, it was clearly inconsistent - so fixing that inconsistency is a good thing.

Which way we fix it, I agree we can discuss. I went with the safer route of keeping the lock. But I suspect it shouldn't be needed here - if we're in this code, it means it's the local player's turn. Which means there shouldn't be concurrent modifications to the state underneath us.

So the only thing this would be preventing is if some changes to the underlying model from another thread are not yet visible to us - but I suspect that shouldn't be possible - since this is happening on the UI thread, which should have gotten current state updates by virtue of transitioning to the move phase and having the map get redrawn, etc.

So I speculate that it should be fine to remove - although I don't know for sure.